packages/core/mesh/network-manager/src/swarm/peer.ts

```diff
@@ -166,7 +173,7 @@ export class Peer {
         this._callbacks.onRejected();
         return;
       }
-      connection.openConnection();
+      await connection.openConnection();
```

Comment:
perhaps set connection = null in closeConnection so that the connection doesnt get closed twice

Reply:
also, any chance there's a race condition?

Reply:
We have an invariant check to avoid that race condition https://github.com/dxos/dxos/blob/main/packages/core/mesh/network-manager/src/swarm/peer.ts